libs/feature-workspaces/src/lib/new-workspace/new-workspace.component.ts

```diff
@@ -51,139 +48,106 @@ interface SchematicCollectionForNgNew {
-    private readonly matDialog: MatDialog,
+    private readonly elementRef: ElementRef,
+    private readonly router: Router,
+    private readonly dialogRef: MatDialogRef<NewWorkspaceComponent>,
+    private readonly fb: FormBuilder,
     private readonly finderService: Finder,
-    private readonly schematicCollectionsGQL: SchematicCollectionsGQL
+    private readonly schematicCollectionsGQL: SchematicCollectionsGQL,
+    private readonly workspacesService: WorkspacesService,
+    private readonly ngNewGQL: NgNewGQL,
+    private readonly commandRunner: CommandRunner
   ) {}
 
-  private static newFormGroup(finderService: Finder): FormGroup {
-    return new FormGroup({
-      path: new FormControl(null, Validators.required),
-      name: new FormControl(
-        null,
-        Validators.required,
-        makeNameAvailableValidator(finderService)
-      ),
-      collection: new FormControl(null, Validators.required)
-    });
+  handleSelection(event: MatSelectionListChange) {
+    if (event.option.selected) {
+      event.source.deselectAll();
+      event.option._setSelected(true);
+      this.ngNewForm.controls.collection.setValue(event.option.value);
+    } else {
+      this.ngNewForm.controls.collection.setValue(null);
+    }
   }
 
-  ngOnInit() {
-    this.contextActionService.contextualActions$.subscribe(actions => {
-      if (!actions) {
-        this.matExpansionPanels.forEach((panel: MatExpansionPanel) => {
-          panel.close();
-        });
-        this.selectedNode = null;
-        this.ngNewForm$.next(
-          NewWorkspaceComponent.newFormGroup(this.finderService)
-        );
-      }
-    });
-
-    this.schematicCollectionsForNgNew$ = this.schematicCollectionsGQL
-      .fetch()
-      .pipe(
-        map(r => r.data.schematicCollections),
-        publishReplay(1),
-        refCount()
-      );
-
-    this.schematicCollectionsForNgNew$.subscribe(() => {
-      this.ngNewForm$.next(
-        NewWorkspaceComponent.newFormGroup(this.finderService)
-      );
-    });
-
-    this.createNewWorkspace$.subscribe(() => {
-      const form = this.ngNewForm$.value;
-      if (form) {
-        this.createNewWorkspace(form.value as NgNewInvocation);
-      }
-    });
-
-    this.ngNewForm$
-      .pipe(
-        filter(form => Boolean(form)) as OperatorFunction<
-          FormGroup | null,
-          FormGroup
-        >,
-        switchMap((form: FormGroup) => form.statusChanges)
-      )
-      .subscribe((formStatus: any) => {
-        this.contextActionService.contextualActions$.next({
-          contextTitle: 'Create a New Angular Workspace',
-          actions: [
-            {
-              name: 'Create',
-              disabled: new BehaviorSubject(formStatus !== 'VALID'),
-              invoke: this.createNewWorkspace$
-            }
-          ]
-        });
+  selectParentDirectory() {
+    this.workspacesService
+      .selectDirectoryForNewWorkspace()
+      .subscribe(result => {
```

Comment:
Do we need an unsubscribe for this?

Reply:
Added pipe to selectDirectoryForNewWorkspace() itself